pkg/apis/core/v1/helper/helpers.go

```diff
@@ -455,6 +455,26 @@ func FindMatchingUntoleratedTaint(taints []v1.Taint, tolerations []v1.Toleration
 	return v1.Taint{}, false
 }
 
+// FindMatchingUntoleratedTaintWithoutTolerationSeconds filters the given tolerations
+// with TolerationSeconds and check if they tolerate all the filtered taints, and
+// returns the first taint without a toleration
+func FindMatchingUntoleratedTaintWithoutTolerationSeconds(taints []v1.Taint, tolerations []v1.Toleration, inclusionFilter taintsFilterFunc) (v1.Taint, bool) {
+	filteredTaints := getFilteredTaints(taints, inclusionFilter)
+	filteredTolerations := []v1.Toleration{}
+	for _, toleration := range tolerations {
+		if toleration.Effect == v1.TaintEffectNoExecute && toleration.TolerationSeconds != nil {
```

Comment:
`tolerationSeconds` only applies to `NoExecute` taints, so I think this section could be simplified to:

```
for _, toleration := range tolerations {
  if toleration.TolerationSeconds == nil {
    filteredTolerations = append(filteredTolerations, toleration)
  }
}
```

As this will capture all `NoSchedule` tolerations and all `NoExecute` taints without a tolerationSeconds set